builtin/providers/aws/resource_aws_sns_topic.go

```diff
@@ -163,6 +165,12 @@ func resourceAwsSnsTopicRead(d *schema.ResourceData, meta interface{}) error {
 	})
 
 	if err != nil {
+		if ec2err, ok := err.(awserr.RequestFailure); ok && ec2err.StatusCode() == http.StatusNotFound {
```

Comment:
Usually (at least from what I've seen elsewhere in the codebase) we check `ec2err.Code()` (which in this case would be `NotFound`) which would help you avoiding that extra import of `net/http`.

Also (but that's really a nitpick), this is a generic AWS error, has nothing to do with EC2. I know this pattern is sprinkled throughout the codebase with EC2 in the name which makes me think we should probably rename all of those variables.